components/brave_ads/core/browser/search_result_ad/search_result_ad_parsing.cc

```diff
@@ -195,12 +195,12 @@ bool SetSearchAdProperty(const schema_org::mojom::PropertyPtr& ad_property,
 
 // TODO(https://github.com/brave/brave-browser/issues/25971): Reduce cognitive
 // complexity.
-absl::optional<SearchResultAdMap> ParseSearchResultAdMapEntityProperties(
```

Comment:
could we consider move all this parsing logic into a sandboxed process or something to comply with https://chromium.googlesource.com/chromium/src/+/master/docs/security/rule-of-2.md ? Since it is coming originally from an "untrusted" context (web page).

Reply:
Thanks for pointing this out! Yes, the parsing logic can be moved to Brave Ads Utility process. I will do this change.

Reply:
@diracdeltas is that relevant here because it has already been serialized and deserialized in mojo?

Reply:
<img width="453" alt="image" src="https://user-images.githubusercontent.com/34129/209240574-d8e1a9cb-0bf5-40c2-9c7f-5ee560dba678.png">

Reply:
also `That way, the higher-privileged process need only process objects adhering to a well-defined, generally low-complexity grammar. This is a big part of why [we like for Mojo messages to use structured types](https://chromium.googlesource.com/chromium/src/+/refs/heads/main/docs/security/mojo.md#Use-structured-types).`

Reply:
Renamed `ParseWebPageEntities` to `ConvertWebPageToSearchResultAds` as discussed in DM.